test/Generics/protocol_requirement_signatures.swift

```diff
@@ -0,0 +1,40 @@
+// RUN: %target-typecheck-verify-swift -typecheck %s -verify
+// RUN: %target-typecheck-verify-swift -typecheck -debug-generic-signatures %s > %t.dump 2>&1
+// RUN: %FileCheck %s < %t.dump
+
+// CHECK-LABEL: .P1@
+// CHECK-NEXT: Requirement signature: <Self where Self : P1>
```

Comment:
This still looks wrong to me. We shouldn't have the `Self: P1` requirement, right?